core/src/wallet/cosmos/api_impl/CosmosLikeTransactionApi.cpp

```diff
@@ -340,5 +340,99 @@ namespace ledger {
+                    auto res = rapidjson::Value(rapidjson::kStringType);
+                    res.SetString(str.c_str(), static_cast<rapidjson::SizeType>(str.size()), allocator);
+                    return res;
+                };
+
+                const auto msgSend = CosmosLikeMessage::unwrapMsgSend(tmpMessage);
+
+                buffer = makeStringValue(msgSend.fromAddress);
+                baseReq.AddMember(cosmos::constants::kFrom, buffer, allocator);
+
+                buffer = makeStringValue(_txData.memo);
+                baseReq.AddMember(cosmos::constants::kMemo, buffer, allocator);
+
+                buffer = makeStringValue(networks::getCosmosLikeNetworkParameters(_currency.name).ChainId);
+                baseReq.AddMember(cosmos::constants::kChainId, buffer, allocator);
+
+                buffer = makeStringValue(_accountNumber);
+                baseReq.AddMember(cosmos::constants::kAccountNumber, buffer, allocator);
+
+                buffer = makeStringValue(_accountSequence);
+                baseReq.AddMember(cosmos::constants::kSequence, buffer, allocator);
+
+                buffer = makeStringValue(_txData.fee.gas.toString());
+                baseReq.AddMember(cosmos::constants::kGas, buffer, allocator);
+
+                // todo : remove duplicated code
+                auto getAmountObject = [&] (const std::string &denom, const std::string &amount) {
+                    auto tmp = rapidjson::Value(rapidjson::kStringType);
+                    auto amountObject = rapidjson::Value(rapidjson::kObjectType);
+                    tmp.SetString(amount.c_str(), static_cast<rapidjson::SizeType>(amount.length()), allocator);
+                    amountObject.AddMember(cosmos::constants::kAmount, tmp, allocator);
+                    tmp.SetString(denom.c_str(), static_cast<rapidjson::SizeType>(denom.length()), allocator);
+                    amountObject.AddMember(cosmos::constants::kDenom, tmp, allocator);
+                    return amountObject;
+                };
+
+                auto fees = rapidjson::Value(rapidjson::kArrayType);
+                auto feeAmount = getAmountObject(_txData.fee.amount[0].denom, _txData.fee.amount[0].amount);
+                fees.PushBack(feeAmount, allocator);
+                baseReq.AddMember(cosmos::constants::kFees, fees, allocator);
+
+                auto simulate = rapidjson::Value(rapidjson::kTrueType);
+                baseReq.AddMember(cosmos::constants::kSimulate, simulate, allocator);
+
+                document.AddMember(cosmos::constants::kBaseReq, baseReq, allocator);
+
+
+                auto amount = rapidjson::Value(rapidjson::kArrayType);
+                auto amountObj = getAmountObject(msgSend.amount[0].denom, msgSend.amount[0].amount);
+                amount.PushBack(amountObj, allocator);
+                document.AddMember(cosmos::constants::kAmount, amount, allocator);
+
+                break;
```

Comment:
Once this works, we'll have to refactor it, probably in static `CosmosLikeMessage` functions, it gets hard to read fast otherwise :)

Reply:
Same than the comment above, I will remove this